### DISCUSSION TITLE
Failing test in smart contract lottery testEmitsEventOnEntrance()

### DISCUSSION THREAD

**Discussion by Raj-RR1** [Cyfrin/foundry-full-course-cu#284] [Q&A]
Hey! I was following the smart contract lottery code specifically the testing part. There is a function testEmitsEventOnEntrance(), whenever i run the forge test command for it i get the below error. I am pretty sure i am sending enough ETH.
```
[FAIL. Reason: Expected an emit, but no logs were emitted afterward. You might have mismatched events or not enough events were emitted.] testEmitsEventOnEntrance() (gas: 14275)
Traces:
  [14275] RaffleTest::testEmitsEventOnEntrance() 
    ├─ [0] VM::prank(player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C]) 
    │   └─ ← ()
    ├─ [0] VM::expectEmit(true, false, false, false, Raffle: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496]) 
    │   └─ ← ()
    ├─ emit EnteredRaffle(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    ├─ [186] Raffle::enterRaffle() 
    │   └─ ← "Raffle__NotEnoughETHSent()"
    └─ ← "Log != expected log"
```

check

**Comment by shikhar229169** [comment]
Actually, the error is due to mismatch in the event emitted.
From the event the contract emitted, differs from what you expected.
So, can you please share the contract event and the test.

**Reply by deep0072** [reply]
thanks

**Comment by Raj-RR1** [comment]
Hey! I looked through my code further and turns out in the  setUp() function in Raffletest i did a mistake of writing data types while destructuring the active network config.

```
        (
            uint256  entranceFee,
            uint256 interval,
            address vrfCoordinator,
             bytes32 gasLane,
             uint64 subscriptionId,
         uint32 callbackGasLimit
        ) = helperConfig.activeNetworkConfig();
```
which was causing this error

**Comment by deep0072** [comment]
event we declared in raffleContract should match with event declared in test File.
